sigstore/_internal/rekor/client.py

```diff
@@ -48,13 +49,47 @@
 
 @dataclass(frozen=True)
 class RekorEntry:
-    uuid: str
+    uuid: Optional[str]
```

Comment:
cc @asraa: I had to make this optional, since offline Rekor bundles don't include their entry UUID. Just a heads up, since we talked a bit about the inclusion of this field before 🙂

Reply:
(The other option here is to do some subclassing, i.e. an abstract `RekorEntry` and `OnlineEntry` and `OfflineEntry` as subclasses.)

Reply:
I'd vote for optional! 

I suppose the UUID information should just be informational (e.g. the hash can be reconstructed from the entry content itself, and the tree ID should just be a hint on which tree the entry is)